db/db_impl/db_impl_open.cc

```diff
@@ -92,9 +92,11 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src) {
     result.wal_recovery_mode = WALRecoveryMode::kTolerateCorruptedTailRecords;
   }
 
+  bool separate_wal_dir = true;
   if (result.wal_dir.empty()) {
     // Use dbname as default
     result.wal_dir = dbname;
+    separate_wal_dir = false;
```

Comment:
It's also possible that user provides a options.wal_dir which is the same as dbname. There might also be added twice, and I don't know the consequences for that.

Reply:
I would compare the wal_dir with db_path but we would need a whole logic to skip extra slash, etc.

Reply:
Taking look at the code it seems fine to double delete a file but given that guarantees of the internal APIs is not cleanly specified it is hard to be confident.

Reply:
Can we at least still set separate_wal_dir if `result.wal_dir == dbname`? I know the byte-wise equality is not sufficient, but should be better than not doing it.

Btw, what's the behavior of double deleting?

Reply:
My comment above says that it is hard to be confident about double deleting.
Sure. byte-wise comparison sounds good.